modules/core/src/v2/wallet.ts

```diff
@@ -2249,4 +2270,112 @@ export class Wallet {
+
+      if (params.reqId) {
+        self.bitgo.setRequestTracer(params.reqId);
+      }
+
+      // this could return 100 build transactions
+      const buildResponse = yield self.bitgo.post(self.baseCoin.url('/wallet/' + self.id() + '/consolidateAccount/build'))
+        .send(whitelistedParams)
+        .result();
+
+      // we need to step over each prebuild now - should be in an array in the body
+      const consolidations:TransactionPrebuild[] = [];
+      for (const consolidateAccountBuild of buildResponse) {
+        let prebuild: TransactionPrebuild = yield self.baseCoin.postProcessPrebuild(
+          Object.assign(consolidateAccountBuild, { wallet: self, buildParams: whitelistedParams })
+        );
+
+        delete prebuild.wallet;
+        delete prebuild.buildParams;
+
+        prebuild = _.extend({}, prebuild, { walletId: self.id() });
+        debug('final consolidation transaction prebuild: %O', prebuild);
+
+        consolidations.push(prebuild);
+      }
+
+      return consolidations;
+    }).call(this).asCallback(callback);
+  }
+
+  /**
+   * Builds and sends a set of consolidation transactions for a wallet.
+   * @param params
+   *     fromAddresses - these are the on-chain receive addresses we want to pick a consolidation amount from
+   */
+  sendAccountConsolidation(params: PrebuildAndSignTransactionOptions = {}, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co<any>(function *() {
+      // one of a set of consolidation transactions
+      if (typeof params.prebuildTx === 'string' || params.prebuildTx === undefined) {
+        throw new Error('Invalid build of account consolidation.');
+      }
+
+      if (!params.prebuildTx.consolidateId) {
+        throw new Error('Failed to find consolidation id on consolidation transaction.');
+      }
+
+      const signedPrebuild = yield self.prebuildAndSignTransaction(params);
+
+      // decorate with our consolidation id
+      signedPrebuild.consolidateId = params.prebuildTx.consolidateId;
+
+      delete signedPrebuild.wallet;
+
+      return yield self.submitTransaction(signedPrebuild);
+    }).call(this).asCallback(callback);
+  }
+
+  /**
+   * Builds and sends a set of account consolidations. This is intended to flush many balances to the root wallet balance.
+   * @param params
+   * @param callback
+   */
+  sendAccountConsolidations(params: BuildConsolidationTransactionOptions = {}, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+
+    if (!this.baseCoin.allowsAccountConsolidations()) {
+      throw new Error('This coin does not allow account consolidations.');
+    }
+
+    return co<any>(function *() {
+      // this gives us a set of account consolidation transactions
+      const unsignedBuilds = yield self.buildAccountConsolidations(params);
+      if (unsignedBuilds && unsignedBuilds.length > 0) {
+        const successfulTxes = new Array<any>();
```

Comment:
Not wrong, but `new Array<any>` is a bit odd, why not `const successfulTxs: any[] = [];`

Reply:
I can change this. I just like explicitly calling the constructor for any new object, it's more of a style thing